moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -2130,10 +2125,9 @@ std::vector<double> moveit::planning_interface::MoveGroupInterface::getCurrentRP
       if (lm)
       {
         result.resize(3);
-        tf::Matrix3x3 ptf;
-        tf::matrixEigenToTF(current_state->getGlobalLinkTransform(lm).rotation(), ptf);
-        tfScalar pitch, roll, yaw;
-        ptf.getRPY(roll, pitch, yaw);
+        geometry_msgs::TransformStamped tfs = tf2::eigenToTransform(current_state->getGlobalLinkTransform(lm));
```

Comment:
Well that's nifty. I'll update with your Eigen method.

Reply:
@rhaschke wouldn't this line store the values in `result` as `[yaw, pitch, roll]` when `result` is expecting `[roll, pitch, yaw]`?

Reply:
Damn. Haven't seen that. Indeed, this would required to switch roll and yaw or to `reverse()` the result of `eulerAngles()`:
`Eigen::Map<Eigen::Vector3d>(&result[0]) = current_state->getGlobalLinkTransform(lm).rotation().eulerAngles(2, 1, 0).reverse()`

However, I'm completely fine, if you decide to keep your code for better readibility.

Reply:
I'll use your updated Eigen method. I think a comment will suffice for readability.

Reply:
After fiddling with this for a bit, I think I will keep the original `tf2` conversion code.